lib/Compilation.js

```diff
@@ -362,13 +362,13 @@ class Compilation extends Tapable {
 	_addModuleChain(context, dependency, onModule, callback) {
 		const start = this.profile && Date.now();
 
-		const errorAndCallback = this.bail ? function errorAndCallback(err) {
+		const errorAndCallback = this.bail ? (err) => {
 			callback(err);
-		} : function errorAndCallback(err) {
```

Comment:
Looking back on the blame for this I believe the reasoning was being a named function. Could we preserve the name for debugging and traces through outer scope assignment etc?